presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingHdfsFileInputStream.java

```diff
@@ -15,36 +15,64 @@
 
 import alluxio.client.file.FileInStream;
 import alluxio.exception.ExceptionMessage;
+import org.apache.hadoop.fs.FSDataInputStream;
 import org.apache.hadoop.fs.PositionedReadable;
 import org.apache.hadoop.fs.Seekable;
 
 import java.io.EOFException;
 import java.io.IOException;
 import java.io.InputStream;
 
+import static com.google.common.base.Verify.verify;
+import static java.util.Objects.requireNonNull;
+
 public class AlluxioCachingHdfsFileInputStream
         extends InputStream
         implements Seekable, PositionedReadable
 {
     private final FileInStream inputStream;
+    private final FSDataInputStream originInputStream;
+    private final boolean cacheValidationEnabled;
 
     public AlluxioCachingHdfsFileInputStream(FileInStream inputStream)
     {
-        this.inputStream = inputStream;
+        this(inputStream, null, false);
+    }
+
+    public AlluxioCachingHdfsFileInputStream(
+            FileInStream inputStream,
+            FSDataInputStream originInputStream,
```

Comment:
nit: Can you rename originInputStream to dataTierInputStream? We are using dataTier to represent the origin source, makes it easy to think what it is.